plugins/org.eclipse.jst.j2ee/archiveops/org/eclipse/jst/j2ee/internal/archive/JavaEEArchiveUtilities.java

```diff
@@ -948,7 +956,8 @@ public boolean isEJBArchive(IArchive archive) {
 										IAnnotation[] annotations = annotationsAttribute.getAnnotations();
 										for (IAnnotation annotation : annotations) {
 											char[] typedName = annotation.getTypeName();
-											if (Arrays.equals(typedName, STATELESS) || Arrays.equals(typedName, STATEFUL) || Arrays.equals(typedName, MESSAGEDRIVEN) || Arrays.equals(typedName, SINGLETON)) {
+											if (Arrays.equals(typedName, STATELESS) || Arrays.equals(typedName, STATEFUL) || Arrays.equals(typedName, MESSAGEDRIVEN) || Arrays.equals(typedName, SINGLETON)
+											    || hasJakartaEJBAnnotation(typedName)) {
```

Comment:
You could introduce a similar method for Java EE to improve readability. However, I'm not sure which name would be suitable. Here are some proposals: `hasJavaEEEJBAnnotation`, `hasOldStyleEJBAnnotation`, `hasLegacyEJBAnnotation`

Reply:
helper Method introduced

Reply:
Generally I've been following a pattern of inserting Jakarta into the name of the parallel new helper methods and constants, e.g. hasEJBAnnotation gets a hasJakartaEJBAnnotation created for that case.

Reply:
There are now two helper methods:
* hasJavaEEEJBAnnotation (legacy annotations for history)
* hasJakartaEJBAnnotation (current Java EE Annotations)
Or statement using this two methods may be switched in future releases for performance optimization (when generally more jakarta.ejb than javax.ejb annotations are used...)